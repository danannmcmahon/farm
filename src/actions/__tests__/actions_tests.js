import configureMockStore from 'redux-mock-store';
import { fetchResultsSuccess } from '../index';

const mockStore = configureMockStore();
const store = mockStore({});

describe('fetch results Success', () => {
    it('creates UPRES when dispatched by fetch results', () => {
      store.dispatch(fetchResultsSuccess(
        {
            "name":"John",
            "category":"sheep",
            "age":6,
            "output":[
                {"product":"wool","frequency":"yearly","quantity":5.3},
                {"product":"meat","frequency":"once","quantity":23.72}],
            "life-expectancy":"12"
        }
      ));
      expect(store.getActions()).toMatchSnapshot();
    });
  });
  