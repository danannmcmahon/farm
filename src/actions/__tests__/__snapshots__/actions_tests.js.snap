// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch results Success creates UPRES when dispatched by fetch results 1`] = `
Array [
  Object {
    "payload": Object {
      "age": 6,
      "category": "sheep",
      "life-expectancy": "12",
      "name": "John",
      "output": Array [
        Object {
          "frequency": "yearly",
          "product": "wool",
          "quantity": 5.3,
        },
        Object {
          "frequency": "once",
          "product": "meat",
          "quantity": 23.72,
        },
      ],
    },
    "type": "UPRES",
  },
]
`;
